models/classes/runner/features/TestRunnerFeature.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+namespace oat\taoTests\models\runner\features;
+
+use oat\oatbox\PhpSerializable;
+use oat\taoTests\models\runner\plugins\TestPlugin;
+
+/**
+ * A test runner feature is a user feature that can be expressed by one or more test runner plugins.
+ * They can be toggled at the delivery level and work only in the new test runner.
+ *
+ * @author Christophe Noël <christophe@taotesting.com>
+ */
+
+abstract class TestRunnerFeature implements PhpSerializable
+{
+    /**
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @var string[] must match active test runner plugins Ids
+     */
+    private $pluginsIds;
+
+    /**
+     * @var bool Determine if the feature will be automatically enabled upon delivery creation
+     */
+    private $isEnabledByDefault;
+
+    /**
+     * @var $allPlugins Used to check the existence of plugins Ids
+     */
+    private $allPlugins;
+
+    /**
+     * @param string $id
+     * @param string[] $pluginsIds
+     * @param bool $isEnabledByDefault
+     * @param TestPlugin[] $allPlugins
+     */
+    public function __construct(
+        $id,
+        $pluginsIds,
+        $isEnabledByDefault,
+        $allPlugins)
+    {
+        $this->id = $id;
+        $this->pluginsIds = $pluginsIds;
+        $this->isEnabledByDefault = $isEnabledByDefault;
+        $this->allPlugins = $allPlugins;
```

Comment:
You should check the provided parameters before assigning them to internal properties. I know the private method `checkData()` will check some of them, but not all are secured.